projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md

```diff
@@ -0,0 +1,103 @@
+# Using Entity Adapter with Feature Creator
+
+This example demonstrates a User state management using NgRx Store with `Entity Adapters` and `Extra Selectors` for a feature.
+
+Usage:
+
+Start by defining the state for the `User` feature.
+
+<code-example header="users.state.ts">
+
+import { EntityState, createEntityAdapter } from '@ngrx/entity';
+import { User } from './user.model';
+
+export interface State extends EntityState&lt;User&lt; {
+  selectedUserId: string | null;
+}
+
+const adapter = createEntityAdapter&lt;User&lt;();
+
+export const initialState: State = adapter.getInitialState({
+  selectedUserId: null,
+});
+
+</code-example>
+
+Then, we define the `User` actions: `addUser` and `selectUser`.
+
+<code-example header="user.actions.ts">
+
+import { createActionGroup, props } from '@ngrx/store';
+import { User } from './user.model';
+
+export const UserListPageActions = createActionGroup({
+  source: 'User List Page',
+  events: {
+    addUser: props&lt;{ user: User }&gt;(),
+    selectUser: props&lt;{ userId: string }&gt;(),
+  },
+});
+
+</code-example>
+
+Then use the `createReducer` function from NgRx to create a reducer for the `User` feature.
+
+<code-example header="users.state.ts">
+
+import { createReducer, on } from '@ngrx/store';
+import { UserListPageActions } from './user-list-page.actions';
+const reducer = createReducer(

```

Comment:
We can also add an empty line between imports and reducer. The rest of the code snippets look great! :)

```suggestion
import { UserListPageActions } from './user-list-page.actions';

const reducer = createReducer(

```